### PR TITLE
Pages

### LINKED ISSUE #7: Add next and previous page buttons to the header.

### PR DESCRIPTION
**Pull request by Peppe**
Adds controls to the header to change pages, to jump directly to a page, and see current page and total number of pages.

Please review #11 and #13 before reviewing this one. 

Fixes #7, #8

### CODE REVIEW

File: demo/pdf-viewer-basic-demos.html [hunk 1]
```diff
@@ -27,6 +27,24 @@ <h3>No File Specified</h3>
...
+    <vaadin-demo-snippet id="pdf-viewer-basic-demos-document-loaded" when-defined="vaadin-pdf-viewer">
+      <template preserve-content>
+        <vaadin-pdf-viewer src="pdf-files/example-invoice.pdf"></vaadin-pdf-viewer>
+        <div id="status">Loading...</div>
+        <script>
+          window.addDemoReadyListener('#pdf-viewer-basic-demos-document-loaded', function (document) {
+            var start = new Date().getTime();
+            const viewer = document.querySelector('vaadin-pdf-viewer');
+            viewer.addEventListener('document-loaded', function (e) {
+
```
[the remark this answers is not in the document]
Comment (Peppe) [reply demo/pdf-viewer-basic-demos.html]: Hi. This PR builds upon #13. This change was done over there. Could you move the change request over there? Then when #13 is accepted, I will rebase this change on top of that.

File: src/vaadin-pdf-viewer.html [hunk 2]
```diff
@@ -81,10 +108,56 @@
             /**
              * A represenentation of a document that has been read in. 
              */
-            _document: Object
+            _document: Object,
```
[the remark this answers is not in the document]
Comment (Peppe) [reply src/vaadin-pdf-viewer.html]: This PR builds upon #13. This change was done over there. Could you move the change request over there? Then when #13 is accepted, I will rebase this change on top of that.

File: src/vaadin-pdf-viewer.html [hunk 3]
```diff
@@ -81,10 +108,56 @@
...
+             * Total amount of pages in an opened document
+             */
+            _totalPages: Number
+
           };
         }
 
+        static get observers() {
+          return [
+            'setTitle(_pdfTitle, _filename)'
```
[the remark this answers is not in the document]
Comment (Peppe) [reply src/vaadin-pdf-viewer.html]: This PR builds upon #13. This change was done over there. Could you move the change request over there? Then when #13 is accepted, I will rebase this change on top of that.

File: src/vaadin-pdf-viewer.html [hunk 4]
```diff
@@ -81,10 +108,56 @@
...
+          ];
+        }
+
+        setTitle(_pdfTitle, _filename) {
```
[the remark this answers is not in the document]
Comment (Peppe) [reply src/vaadin-pdf-viewer.html]: This PR builds upon #13. This change was done over there. Could you move the change request over there? Then when #13 is accepted, I will rebase this change on top of that.

File: theme/material/vaadin-pdf-viewer.html [hunk 5]
```diff
@@ -1,15 +1,81 @@
...
+      [part~="toolbar-button"]:hover {
+        opacity: 1;
+      }
+
+      [part~="toolbar-button"]:disabled:hover {
+        opacity: 0.4;
+      }
+
+      [part~="toolbar-button"] {
+        font-family: 'lumo-icons';
```
Comment (tomivirkki) [inline theme/material/vaadin-pdf-viewer.html]: Drop the lumo-icons from Material theme

File: src/vaadin-pdf-viewer.html [hunk 6]
```diff
@@ -95,6 +168,12 @@
           this.$.viewerContainer.addEventListener('pagesinit', () => {
             this._viewer.currentScaleValue = 'auto';
           });
+          this.$.viewerContainer.addEventListener('pagechange', (event) => {
+            this._currentPage = event.pageNumber;
+            this._updatePageNumberStates();
+          });
+          pdfjsLib.GlobalWorkerOptions.workerSrc =
+            '../../pdfjs-dist/build/pdf.worker.min.js';
           this.open(this.src);
```
Comment (tomivirkki) [inline src/vaadin-pdf-viewer.html]: `src` is public API but it's only read once (on ready()). Shouldn't you be able to dynamically set the component `src` at any time? You'll need a property observer for that

File: src/vaadin-pdf-viewer.html [hunk 7]
```diff
@@ -111,14 +190,89 @@
...
+              detail: {
+                document: pdfDocument
+              }
+            }));
           }, function (exception) {
             console.error(exception && exception.message);
           });
         }
+
+        setFilename(src) {
```
Comment (tomivirkki) [inline src/vaadin-pdf-viewer.html]: Is this API intentionally public?

File: src/vaadin-pdf-viewer.html [hunk 8]
```diff
@@ -111,14 +190,89 @@
...
+          try {
+            filename = decodeURIComponent(filename);
+          } catch (e) {
+            // decodeURIComponent may throw URIError,
+            // fall back to using the unprocessed url in that case
+          }
+          this._filename = filename;
+        }
+
+        setPdfTitleFromMetadata(pdfDocument) {
```
Comment (tomivirkki) [inline src/vaadin-pdf-viewer.html]: Is this API intentionally public?

File: src/vaadin-pdf-viewer.html [hunk 9]
```diff
@@ -111,14 +190,89 @@
...
+
+            const info = data.info;
+            if (!pdfTitle && info && info['Title']) {
+              pdfTitle = info['Title'];
+            }
+            self._pdfTitle = pdfTitle;
+          });
+        }
+
+        _updatePageNumberStates() {
```
Comment (tomivirkki) [inline src/vaadin-pdf-viewer.html]: Is this API intentionally protected (use __ for private)?

Same goes for the methods below

File: src/vaadin-pdf-viewer.html [hunk 10]
```diff
@@ -111,14 +190,89 @@
...
+            page = this._totalPages;
+          }
+          this._viewer.currentPageNumber = page;
+        }
+
+        _getPage(){
+          return this._viewer.currentPageNumber;
+        }
+
+        previousPage() {
```
Comment (tomivirkki) [inline src/vaadin-pdf-viewer.html]: Why not instead expose `currentPage` property that allows for example to jump to any page. These methods only allow navigating one page at a time.

File: theme/material/vaadin-pdf-viewer.html (hunk 5 above)
Comment (Peppe) [reply theme/material/vaadin-pdf-viewer.html]: Done.

File: src/vaadin-pdf-viewer.html (hunk 7 above)
Comment (Peppe) [reply src/vaadin-pdf-viewer.html]: No. Fixed.

File: src/vaadin-pdf-viewer.html (hunk 8 above)
Comment (Peppe) [reply src/vaadin-pdf-viewer.html]: No. Fixed.

File: src/vaadin-pdf-viewer.html (hunk 9 above)
Comment (Peppe) [reply src/vaadin-pdf-viewer.html]: Changed to private. Done.

File: src/vaadin-pdf-viewer.html (hunk 10 above)
Comment (Peppe) [reply src/vaadin-pdf-viewer.html]: Moved these to private.

**Pull request merged**